### ISSUE TITLE
Users can't connect to dapp or it takes forever

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#5509]
This issue is unresolved
https://github.com/leather-wallet/extension/issues/5021

I see these errors but I am not sure if they are related. It his happening on all dapps. When account data isn't fully loaded yet. It takes forever even when the balance is loaded. This happening especially in wallets with 20 or more accounts and when for example the account balance is still being loaded. I have also seen the same issue (connection to dapp doesn't complete) when balance was just loaded. 

Steps to reproduce
0.  Restore Leather with an existing Secret Key that has 20 or more accounts in it.
1. Click connect on app.alexlab.co
or stacks.gamma.io
or any other dapp
2. Scroll down until you see all the "loading" animations for balance
3. Click one of the accounts with the accounts to connect
4. spinning (to connect) starts and never stops... dapps never connects
![image](https://github.com/leather-wallet/extension/assets/33360391/31e9ac67-926f-4643-8119-e5356edad82a)

Possibly unrelated console errors
![image](https://github.com/leather-wallet/extension/assets/33360391/621a56bb-5355-4567-86a8-e3e973bc595c)

Workarounds
- Wait longer for the balances in the "select account" screen to load (works most of the time)
- Click additional times (sometimes works)

**Comment by 314159265359879** [comment]
It looks like users can't switch to different accounts within Leather either because there is loading or throttling going on. 

In the above example the wallet is stuck when scrolling over to connect to a specific account on a dapp.



The video displays another issue, that is reproducible with these steps:
1. being on the last account in a  wallet, something in the 400 range
2. Scrolling all the way up to account 1 (why is there no scroll bar?)
3. Selecting account 1, clicking it a couple times in the hope of working around the issue... doesn't work
4. click outside the wallet so the modal closes
5. try to open it again... doesn't work
6. Here the extension icon in the browser is clicked and nothing happens (Brave, wallet version 6.42.0):

https://github.com/leather-wallet/extension/assets/33360391/e3b5554a-3a6b-46e8-b4df-b5cd46f1bbf3

**Comment by markmhendrickson** [comment]
@alter-eggo this is quite high priority and presumably related to client-side throttling?

**Comment by alter-eggo** [comment]
@markmhendrickson yes, seems so, will take a look

**Comment by 314159265359879** [comment]
Confirmed fixed in this PR https://github.com/leather-io/extension/pull/5636

**Comment by 314159265359879** [comment]
I see this issue again testing a testnet app
in the list of accounts to connect with, I click an account and the loading spinner keeps spinning for a minute (perhaps longer) before actually proceeding to connect. 

On mainnet it is quicker

**Comment by markmhendrickson** [comment]
@alter-eggo any idea why it may be slower on testnet vs. mainnet given your work on this issue?

**Comment by alter-eggo** [comment]
hm, we have quite tight limiting rule on testnet, I'll fix it

**Comment by markmhendrickson** [comment]
@314159265359879 to confirm if still an issue in prod

**Comment by 314159265359879** [comment]
I am def. seeing this again now I saw it connecting to stacks.gamma.io, connecting to renewbns.com and connecting to this https://hozzjss.github.io/stacks-ogs/friedger on mainnet. 

Here are some of the console errors
![image](https://github.com/user-attachments/assets/fa16e4bb-08fb-4cda-9353-196035384b12)


![image](https://github.com/user-attachments/assets/c919a6eb-bc8d-4696-8f50-899c0c480663)


The errors could be just this last site. On gamma I only see these
![image](https://github.com/user-attachments/assets/b19366ef-1b9c-4926-bbcf-c9f03b993dfd)


I have not been able to test this with the 1.2 sec build.

**Comment by markmhendrickson** [comment]
@314159265359879 want to test this again with https://github.com/leather-io/extension/pull/5726 once https://github.com/leather-io/extension/pull/5690 is merged into that release?

**Comment by 314159265359879** [comment]
It looks better now with 6.45.2 and those PR's included.
